First stab at adding global SmartStore

bhariharan opened the pull request:
TODO Items:

* Change the filter for deletion to exclude global DBs.
* Add new tests.
* Verify existing tests pass.

bhariharan commented on libs/SmartStore/src/com/salesforce/androidsdk/smartstore/app/SalesforceSDKManagerWithSmartStore.java:
```diff
@@ -218,6 +223,25 @@ public synchronized void changePasscode(String oldPass, String newPass) {
+     * 				 filename extension such as ".db". Pass 'null' for default.
+     * @return SmartStore instance.
+     */
+    public SmartStore getGlobalSmartStore(String dbName) {
```
Passing in ```null``` or empty string for ```dbName``` here will use the default name. We could expose another method that takes no parameters if we want to, but there are already 4 or 5 of these types of methods here.

bhariharan commented on libs/SmartStore/src/com/salesforce/androidsdk/smartstore/app/SalesforceSDKManagerWithSmartStore.java:
```diff
@@ -309,4 +347,55 @@ public boolean hasSmartStore(UserAccount account, String communityId) {
+      * @param dbName Database name. This must be a valid file name without a
+      * 			  filename extension such as ".db". Pass 'null' for default.
+      */
+     public void removeGlobalSmartStore(String dbName) {
```
```remove``` was never exposed here before, although it was possible through ```DbOpenHelper```.

bhariharan commented:
Added re-encryption of global DBs in [this](https://github.com/bhariharan/SalesforceMobileSDK-Android/commit/59b6449e2ef383fcafc1afff09dd43f74e329107) commit.

khawkins commented on libs/SmartStore/src/com/salesforce/androidsdk/smartstore/store/DBOpenHelper.java:
```diff
@@ -95,20 +105,19 @@ public static synchronized DBOpenHelper getOpenHelper(Context ctx,
-		StringBuffer dbName = new StringBuffer(dbNamePrefix);
-		
-		// If we have account information, we will use it to create a database suffix for the user
+	public static DBOpenHelper getOpenHelper(Context ctx, String dbNamePrefix,
```
Ah, so like you said, this functionality already existed.

khawkins commented:
Looks good to me, over all.

wmathurin commented:
Looks fine.

bhariharan commented:
Looks like we already had tests for global smart store in ```DbOpenHelperTest```. Nothing for me to do there :-)

bhariharan commented:
See [this](https://github.com/bhariharan/SalesforceMobileSDK-Android/commit/8dda2173f43c8d41137b6b9bf175223315fe2222) commit for the last round of changes.

* Added tests for removal of DBs.
* Fixed the file filter logic, so that global DBs are not removed upon logout.
* Misc. cleanup.

khawkins commented:
:+1:
